public/app/features/dashboard-scene/saving/getDashboardChanges.ts

```diff
@@ -30,24 +35,40 @@ export function isEqual(a: VariableOption | undefined, b: VariableOption | undef
   return a === b || (a && b && a.selected === b.selected && deepEqual(a.text, b.text) && deepEqual(a.value, b.value));
 }
 
-// TODO[schema v2]
 export function getRawDashboardV2Changes(
   initial: DashboardV2Spec,
   changed: DashboardV2Spec,
   saveTimeRange?: boolean,
   saveVariables?: boolean,
   saveRefresh?: boolean
 ) {
+  const initialSaveModel = initial;
+  const changedSaveModel = changed;
+  const hasTimeChanged = getHasTimeChanged(changedSaveModel.timeSettings, initialSaveModel.timeSettings);
+  const hasVariableValueChanges = applyVariableChangesV2(changedSaveModel, initialSaveModel, saveVariables);
```

Comment:
I reused `getHasTimeChanged` but for variable changes, it needs to iterate over all the variables, so I created a new one for V2.